src/framework/ui/icon/iconRegistry.component.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { IconRegistryService } from './service/iconRegistry.service';
+import { IconPack } from './service/type';
+
+type IconsProp = IconPack<any> | IconPack<any>[];
+
+export interface IconRegistryProps {
+  icons: IconsProp;
+  defaultIcons?: string;
+}
+
+/**
+ * Icon Registry component. Registers one or more icon packs for later usage in `<Icon/>` component.
+ * Renders nothing, but should be added as a child of an Application Root.
+ *
+ * @extends React.Component
+ *
+ * @property {IconPack<any> | IconPack<any>[]} icons - Icon packs to register.
+ * @property {string} defaultIcons - Name of an icon pack that is used by default.
+ *
+ * @overview-example Simple Usage
+ *
+ * ```
+ * import React from 'react';
+ * import { mapping, light as lightTheme } from '@eva-design/eva';
+ * import { ApplicationProvider, IconRegistry } from 'react-native-ui-kitten';
+ * import { EvaIconsPack } from '@ui-kitten/eva-icons'; // <-- Make sure it is installed. npm i @ui-kitten/eva-icons
+ * import { Application } from './path-to/root.component';
+ *
+ * export default class App extends React.Component {
+ *
+ *   public render(): React.ReactNode {
+ *     return (
+ *       <ApplicationProvider
+ *         mapping={mapping}
+ *         theme={lightTheme}>
+ *         <Application/>
+ *         <IconRegistry icons={EvaIconsPack}/>
+ *       </ApplicationProvider>
+ *     );
+ *   }
+ * }
+ * ```
+ */
+export class IconRegistry extends React.Component<IconRegistryProps> {
```

Comment:
Should it be the separate component that stores icons-pack? Also, can we change the icons-pack in runtime in this case?

Reply:
Yes
I've gived a detailed explanation on it's usage in a [setup guide](https://github.com/akveo/react-native-ui-kitten/blob/7de76cda8cfbbf7cabdfa8ecd0c7dafdc7e8ad92/docs/src/articles/guides/setup-icons-module.md#configure-icon-registry)

Also, take a look at [supporting multiple icon packs guide](https://github.com/akveo/react-native-ui-kitten/blob/7de76cda8cfbbf7cabdfa8ecd0c7dafdc7e8ad92/docs/src/articles/guides/setup-vector-icons.md)

[Runtime icon pack change](https://github.com/akveo/react-native-ui-kitten/blob/7de76cda8cfbbf7cabdfa8ecd0c7dafdc7e8ad92/docs/src/articles/guides/setup-vector-icons.md#choose-a-pack)